### PR TITLE
stm32g0: Add identification for STM32C011 and STM32C031

### PR DESCRIPTION
**Pull request by djix123**
## Detailed description

Add identification for new STM32C011 and STM32C031 parts. These appear to be simplified version of STM32G03x parts. As such they share their part_id with STM32G03x parts but their DBG_IDCODE differs.

The patch is implemented by adding an additional switch() statement to the STM32G03x identification path. If there is another preferred way of doing this, please let me know and I will adjust.

References: STM reference manuals: RM0444 & RM0490.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/stm32g0.c [hunk 1]
```diff
@@ -206,11 +215,30 @@ bool stm32g0_probe(target_s *t)
 	size_t flash_size = 0U;
 
 	switch (t->part_id) {
-	case STM32G03_4:
-		/* SRAM 8kiB, Flash up to 64kiB */
-		ram_size = RAM_SIZE_G03_4;
-		flash_size = FLASH_SIZE_MAX_G03_4;
-		t->driver = "STM32G03/4";
+	case STM32G03_4:;
+		uint32_t dev_id = target_mem_read32(t, DBG_IDCODE) & 0xfffU;
```
Comment (dragonmux) [inline src/target/stm32g0.c:219]: Please make `dev_id` `const` (`const uint32_t dev_id`)

**Review by dragonmux — CHANGES_REQUESTED** [review]
We spotted one small changes needs making, but otherwise this looks good. We'll merge once that that one issue is fixed

Thanks for the contribution!

File: src/target/stm32g0.c (hunk 1 above)
Comment (djix123) [reply src/target/stm32g0.c]: Thanks! Done - and re-pushed.

**Review by dragonmux — APPROVED** [review]
LGTM, merging

**Pull request merged**